crates/sui-framework/packages/bridge/tests/bridge_setup.move

```diff
@@ -0,0 +1,695 @@
+            env.chain_id,
+            3, // seq_num
+            false, // native_token
+            vector[BTC_ID, ETH_ID, USDC_ID, USDT_ID],
+            vector[
+                type_name::get<BTC>().into_string(),
+                type_name::get<ETH>().into_string(),
+                type_name::get<USDC>().into_string(),
+                type_name::get<USDT>().into_string(),
+            ],
+            vector[1000, 100, 1, 1],
+        );
+        let payload = add_token_message.extract_add_tokens_on_sui();
+        bridge.test_execute_add_tokens_on_sui(payload);
+
+        test_scenario::return_shared(bridge);
+    }
+
+    //
+    // Getters
+    //
+
+    public fun validator_pubkeys(): vector<vector<u8>> {
+        vector[
+            VALIDATOR1_PUBKEY,
+            VALIDATOR2_PUBKEY,
+            VALIDATOR3_PUBKEY,
+        ]
+    }
+
+    public fun ctx(env: &mut BridgeEnv): &mut TxContext {
+        env.scenario.ctx()
+    }
+
+    public fun scenario(env: &mut BridgeEnv): &mut Scenario {
+        &mut env.scenario
+    }
+
+    public fun bridge(env: &mut BridgeEnv, sender: address): BridgeWrapper {
+        let scenario = &mut env.scenario;
+        scenario.next_tx(sender);
+        let bridge = scenario.take_shared<Bridge>();
+        BridgeWrapper { bridge }
+    }
+
+    public fun bridge_ref(wrapper: &BridgeWrapper): &Bridge {
+        &wrapper.bridge
+    }
+
+    public fun return_bridge(bridge: BridgeWrapper) {
+        let BridgeWrapper { bridge } = bridge;
+        test_scenario::return_shared(bridge);
+    }
+
+    public fun get_btc(env: &mut BridgeEnv, amount: u64): Coin<BTC> {
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.btc_coins.split(amount, ctx)
+    }
+
+    public fun get_eth(env: &mut BridgeEnv, amount: u64): Coin<ETH> {
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.eth_coins.split(amount, ctx)
+    }
+
+    public fun get_usdc(env: &mut BridgeEnv, amount: u64): Coin<USDC> {
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.usdc_coins.split(amount, ctx)
+    }
+
+    public fun get_usdt(env: &mut BridgeEnv, amount: u64): Coin<USDT> {
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.usdt_coins.split(amount, ctx)
+    }
+
+    //
+    // Bridge commands
+    //
+
+    // Register new tokens
+    public fun register_foreign_token<T>(
+        env: &mut BridgeEnv,
+        treasury_cap: TreasuryCap<T>,
+        upgrade_cap: UpgradeCap,
+        metadata: CoinMetadata<T>,
+        sender: address,
+    ) {
+        let scenario = env.scenario();
+        scenario.next_tx(sender);
+        let mut bridge = scenario.take_shared<Bridge>();
+        bridge.register_foreign_token<T>(treasury_cap, upgrade_cap, &metadata);
+        test_scenario::return_shared(bridge);
```

Comment:
check waiting_room and treasuries have proper changes here

Reply:
done